soliket/xcorr/limber.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pdb
+
+oneover_chmpc = 1. / 2997.92458
+
+
+def mag_bias_kernel(cosmo, dndz, s1, zatchi, chi_arr, chiprime_arr, zprime_arr):
+
+    dndzprime = np.interp(zprime_arr, dndz[:, 0], dndz[:, 1], left=0, right=0)
+    norm = np.trapz(dndz[:, 1], x=dndz[:, 0])
+    dndzprime = dndzprime / norm #TODO check this norm is right
+
+    g_integrand = (chiprime_arr - chi_arr[np.newaxis, :]) / chiprime_arr \
+                    * (oneover_chmpc * cosmo.get_param('H0') / 100) \
+                    * np.sqrt(cosmo.get_param('omegam') * (1 + zprime_arr)**3.
+                                + 1 - cosmo.get_param('omegam')) \
+                    * dndzprime
+
+    g = chi_arr * np.trapz(g_integrand, x=chiprime_arr, axis=0)
+
+    W_mu = (5. * s1 - 2.) * 1.5 * cosmo.get_param('omegam') \
+                * (cosmo.get_param('H0') / 100)**2 * (oneover_chmpc)**2 \
+                * (1. + zatchi(chi_arr)) * g
+
+    return W_mu
+
+
+def do_limber(ell_arr, cosmo, dndz1, dndz2, s1, s2, pk, b1_HF, b2_HF,
+              alpha_auto, alpha_cross,
+              use_zeff=True, autoCMB=False,
+              Nchi=50, dndz1_mag=None, dndz2_mag=None,
+              normed=False, setup_chi_flag=False, setup_chi_out=None):
+
+    zatchi, chiatz, chi_arr, z_arr, chiprime_arr, zprime_arr = setup_chi_out
```

Comment:
sure this will never happen, but what if do_limber is invoked with setup_chi_out=None by mistake? Would you like to add an error message for that?